ostd/src/exception/mod.rs

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Handles exceptions (including traps, faults or interrupts) across kernel and user space.
+
+pub mod handler;
+mod irq;
+pub mod softirq;
+
+pub use softirq::SoftIrqLine;
+pub use trapframe::TrapFrame;
```

Comment:
`softirq::SoftIrqLine` should not be re-exported in the same fashion as the external API `trapframe::TrapFrame`.

Reply:
I have no opinion on `softirq::SoftIrqLine`, but `trapframe::TrapFrame` is used by the handler API as parameters and used by driver crates. If we do not re-export `trapframe::TrapFrame` the driver crates need to explicitly depend on trapframe. This may lead to inconsistency.